### DISCUSSION TITLE
[mason-lspconfig.nvim] Server "lua_ls" is not a valid entry in ensure_installed.

### DISCUSSION THREAD

**Discussion by gerroon** [LazyVim/LazyVim#2154] [Q&A] [answered]
Hi

So when I open a lua file this is the message I get. I am not a lua dev but some config files are written in lua so thatis why I have to deal with them. THinking that I do not have a lua compiler, I downloaded lua 5.1 (which is the only windows version I could find) and added to the path, but the issue is still there. Not sure what else I need :(

`[mason-lspconfig.nvim] Server "lua_ls" is not a valid entry in ensure_installed. Make sure to only provide lspconfig server names.`

**Comment by dpetka2001** [comment]
What version of `nvim-lspconfig` do you have? Have you updated all the plugins via the `Lazy` UI? When you installed LazyVim did you remove/backup the necessary folders as stated in the [installation](https://www.lazyvim.org/installation) guide or is there a chance you had some leftovers from a previous installation of Neovim?

**Reply by gerroon** [reply]
Here is what I did

Install Nvvim
Install LazyVim
Update LazyVim/plugins in LazyVim 

There were all completely all fresh new installations on my laptop. I can take a look at it to see if there were any other stuff. I am on Windows, which folders should I remove and try again?

**Reply by dpetka2001** [reply]
From inside Neovim run `:echo stdpath("data")`. That will tell you which directory it is.

**Reply by dpetka2001** [reply]
Also what version of Neovim do you use?

**Comment by RickyAtkinson** [comment, marked as answer]
I had this same issue on my Windows box a bit back. If I remember correctly, it was because I didn't have the "unzip" cli tool installed on Windows. Unzip is a dependency for mason. I installed unzip with the scoop package manager and everything worked fine.

Running ":CheckHealth" inside neovim should give you a report that will let you know if unzip or any other dependencies are missing.

Hope this helped. That error message gave me a head ach for a few days...

**Reply by gerroon** [reply]
Hey thanks, that seemed to solve my issue, which is quite weird that I had it.

**Reply by dpetka2001** [reply]
Nice!! I didn't even think about that. I guess i should think about running `:checkhealth` more often in the future. Thank you for your insight @RickyAtkinson.
